### DISCUSSION TITLE
Filtering cell subsets based on expression of a gene

### DISCUSSION THREAD

**Discussion by ASAHanna** [satijalab/seurat#5764] [Q&A]
I would like to ask for your help. Is it possible to manually filter two cell subsets based on the expression of a specific gene, then do differential gene expression analysis?
For example, Can we filter cells with high gene A expression vs cells with low gene A expression, then analyze differential gene expression between these two cell subsets? 
I found `subset(x = my.seurat.object, subset = my.gene >= 4)` but I can't go from here to actually having two subsets and doing gene expression analysis between the two subsets.
Thank you!

**Comment by ASAHanna** [comment]
I figured out a code, and would to ask how to include the new cell identity in the metadata so that I can visualize DE after FindMarkers? 

#subsetting MIfibroblast.obj with "high" Postn gene exp
PostnHigh.obj <- subset(MIfibroblast.obj, subset = Postn > 3)

#change identity of cells in PostnHigh object
PostnHigh.obj <- RenameIdents(object = PostnHigh.obj, `14MI Fibroblast` = "PostnHi")

#subsetting MIfibroblast.obj with "low" Postn gene exp
PostnLow.obj <- subset(MIfibroblast.obj, subset = Postn < 0.1)
PostnLow.obj <- RenameIdents(object = PostnLow.obj, `14MI Fibroblast` = "PostnLo")

#Merging GeneHigh and Low into one obj
Postn.obj <- merge(PostnHigh.obj, y = PostnLow.obj)

#DEG
Postn.markers <- FindMarkers(Postn.obj, ident.1 = "PostnHi", ident.2 = "PostnLo", verbose = FALSE)

**Reply by JIANshi-dot** [reply]
What is "14MI Fibroblast"
